conventional-commits-common/src/main/java/com/smartling/cc4j/semantic/release/common/changelog/ChangelogGenerator.java

```diff
@@ -0,0 +1,118 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ChangelogGenerator {
+    public static final int COMMIT_HASH_DISPLAYED_LENGTH = 10;
+    private final Logger logger = LoggerFactory.getLogger(LogHandler.class);
+
+    private static final String CHANGELOG_FORMAT = "## %s (%s)" +
+        "%s";
+    private static final String MD_LINK_FORMAT = "[%s](%s)";
+    private static final String BUG_FIXES_HEADER = "Bug fixes";
+    private static final String FEATURE_HEADER = "Feature";
+    private static final String BREAKING_HEADER = "Breaking changes";
+    private static final String DOCS_HEADER = "Docs";
+    private static final String CI_HEADER = "CI";
+    private static final String BUILD_HEADER = "Build";
+
+    private final String repoUrlFormat;
+    private final String trackingSystemUrlFormat;
+
+    public ChangelogGenerator(String repoUrlFormat, String trackingSystemUrlFormat) {
+        this.repoUrlFormat = repoUrlFormat;
+        this.trackingSystemUrlFormat = trackingSystemUrlFormat;
+    }
+
+    public String generate(String nextVersion, Map<ConventionalCommitType, Set<Commit>> commitsByCommitType) {
+        Objects.requireNonNull(nextVersion, "next version can not be null");
+        Objects.requireNonNull(commitsByCommitType, "commits by type can not be null");
+
+        List<String> sections = new ArrayList<>();
+
+        if (commitsByCommitType.get(ConventionalCommitType.BREAKING_CHANGE) != null) {
+            getSection(BREAKING_HEADER, commitsByCommitType.get(ConventionalCommitType.BREAKING_CHANGE)).ifPresent(sections::add);
+        }
+
+        if (commitsByCommitType.get(ConventionalCommitType.FIX) != null) {
+            getSection(BUG_FIXES_HEADER, commitsByCommitType.get(ConventionalCommitType.FIX)).ifPresent(sections::add);
+        }
+
+        if (commitsByCommitType.get(ConventionalCommitType.FEAT) != null) {
+            getSection(FEATURE_HEADER, commitsByCommitType.get(ConventionalCommitType.FEAT)).ifPresent(sections::add);
+        }
+
+        if (commitsByCommitType.get(ConventionalCommitType.DOCS) != null) {
+            getSection(DOCS_HEADER, commitsByCommitType.get(ConventionalCommitType.DOCS)).ifPresent(sections::add);
+        }
+
+        if (commitsByCommitType.get(ConventionalCommitType.CI) != null) {
+            getSection(CI_HEADER, commitsByCommitType.get(ConventionalCommitType.CI)).ifPresent(sections::add);
+        }
+
+        if (commitsByCommitType.get(ConventionalCommitType.BUILD) != null) {
+            getSection(BUILD_HEADER, commitsByCommitType.get(ConventionalCommitType.BUILD)).ifPresent(sections::add);
+        }
+
+        sections = sections.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        return String.format(CHANGELOG_FORMAT, nextVersion, LocalDate.now(), "\n" + String.join("\n", sections));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
Hi @lillieMaiBauer, i added a spot bugs plugin to the project. 
%n - looks the right thing to do. But i realized this case: 
1. CI server uses Unix based OS with \n line separator - changelog contains \n as a line separator
2. For some reason we decided to perform release from Windows host - which will replace \n to \r\n in the changelog file
I am not sure if this is the correct behavior? I can see in many sources that %n should be used, however - is the case described above ok - so we really want to use platform specific separator in generated changelog file?